### PR TITLE
Implementing DBN copy method with tests

### PR DESCRIPTION
**Pull request by aelnouby**
I have implemented the copy method for the DBN class according to #489 , I have also added the unit tests. Please review this and tell me if there are any tests that should be added,

### CODE REVIEW

File: pgmpy/tests/test_models/test_DynamicBayesianNetwork.py [hunk 1]
```diff
@@ -140,6 +140,33 @@ def test_moralize(self):
...
+        self.assertNotEqual(sorted(self.network.nodes()), sorted(copy.nodes()))
+        self.assertNotEqual(sorted(self.network.edges()), sorted(copy.edges()))
+        self.assertNotEqual(self.network.get_cpds(), copy.get_cpds())
+        self.assertNotEqual(sorted(self.network.get_intra_edges()), sorted(copy.get_intra_edges()))
+        self.assertListEqual(sorted(self.network.get_inter_edges()), sorted(copy.get_inter_edges()))
+        self.assertNotEqual(sorted(self.network.get_slice_nodes()), sorted(copy.get_slice_nodes()))
+        self.network.add_edge(('A', 0), ('D', 1))
+        copy.add_edge(('Z', 0), ('D', 1))
+        self.assertNotEqual(sorted(self.network.get_inter_edges()), sorted(copy.get_inter_edges()))
+
```
Comment (khalibartan) [inline pgmpy/tests/test_models/test_DynamicBayesianNetwork.py:177]: Try changing the values present in CPD, and see if it changes in the original one or not.

Comment (aelnouby) [inline pgmpy/tests/test_models/test_DynamicBayesianNetwork.py:177]: @khalibartan  I actually don't know what method to use to edit CPD values of an existent network, could you tell me how to do this ?

Comment (khalibartan) [inline pgmpy/tests/test_models/test_DynamicBayesianNetwork.py:177]: No need of method just directly use `cpd.values` and update some of it, or change them (cpd values).

**Comment by aelnouby** [comment]
@khalibartan  Check now if this what you were asking about

File: pgmpy/tests/test_models/test_DynamicBayesianNetwork.py (hunk 1 above)
Comment (khalibartan) [inline pgmpy/tests/test_models/test_DynamicBayesianNetwork.py:177]: No, this is different you are reinitializing a variable, I asked you to change values of cpd, not completely .
You have to take a cpd from copy.cpds and change its value.

Comment (aelnouby) [inline pgmpy/tests/test_models/test_DynamicBayesianNetwork.py:177]: @khalibartan you mean something like this ?
`copy.cpds[0].values=np.array([[0.4, 0.05, 0.8, 0.5],[0.3, 0.25, 0.1, 0.3],[0.3, 0.7, 0.1, 0.2]])`

Comment (khalibartan) [inline pgmpy/tests/test_models/test_DynamicBayesianNetwork.py:177]: Yeah that will do.

File: pgmpy/tests/test_models/test_DynamicBayesianNetwork.py [hunk 2]
```diff
@@ -140,6 +140,43 @@ def test_moralize(self):
                               [('D', 1), ('G', 1)], [('D', 1), ('I', 1)],
                               [('G', 0), ('I', 0)], [('G', 1), ('I', 1)]])
 
+    def test_copy(self):
+        self.network.add_edges_from(
+            [(('D', 0), ('G', 0)), (('I', 0), ('G', 0)), (('D', 0), ('D', 1)), (('I', 0), ('I', 1))])
+        cpd = TabularCPD(('G', 0), 3, values=[[0.3, 0.05, 0.8, 0.5],
+                                                         [0.4, 0.25, 0.1, 0.3],
+                                                         [0.3, 0.7, 0.1, 0.2]],
```
Comment (khalibartan) [inline pgmpy/tests/test_models/test_DynamicBayesianNetwork.py]: Run `PEP8` at here. I think this is over indented. You can use following command

```
pep8 <file_name> --max-line-length=120
```

**Comment by khalibartan** [comment]
@ankurankan  changes looks good to me. MRG from my side

**Pull request merged**